src/specfem3D/compute_kernels.F90

```diff
@@ -860,13 +860,25 @@ subroutine compute_kernels_Hessian()
 
   implicit none
 
+  logical, parameter :: USE_SOURCE_RECEIVER_HESSIAN = .False.
+  real(kind=CUSTOM_REAL), dimension(3, 3) :: veloc_deriv, b_veloc_deriv
+  real(kind=CUSTOM_REAL) :: hess_rho, hess_kappa, hess_mu
+
   ! local parameters
   integer :: ispec,iglob
 #ifdef FORCE_VECTORIZATION
   integer :: ijk
-#else
-  integer :: i,j,k
 #endif
+  integer :: i,j,k
+
+  interface
```

Comment:
Hi Daniel, I think I have to define the interface explicitly, otherwise if I delete the interface, the compile will faild:
```
Error: Procedure 'compute_gradient_crust_mantle' at (1) with assumed-shape dummy argument 
'field' must have an explicit interface
```
Since the input argument `field` is defined as
```
real(kind=CUSTOM_REAL), dimension(:, :), intent(in) :: field
```

The other way around would be define the function in a module.

Reply:
right, i missed that you had left field without dimensions. 

we usually assign field sizes like `..,dimension(NDIM,nglob)` and either pass nglob as argument or use the ones saved in the mesher file `values_from_mesher.h` like `NGLOB_CRUST_MANTLE`. that way, we can omit the interface definitions.

Reply:
Thanks. Fixed with explicitly specifying the array dimension.